site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/OrganizationStore.java

```diff
@@ -0,0 +1,59 @@
+package uk.ac.ox.ndph.mts.site_service.repository;
+
+import org.hl7.fhir.r4.model.Organization;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.ResearchStudy;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import uk.ac.ox.ndph.mts.site_service.converter.EntityConverter;
+import uk.ac.ox.ndph.mts.site_service.model.Site;
+
+/**
+ * Implement an EntityStore for Site.
+ */
+@Component
+public class OrganizationStore implements EntityStore<Site> {
+
+
+    private FhirRepository repository;
+    private EntityConverter<Site, org.hl7.fhir.r4.model.Organization> converter;
+
+    /**
+     *
+     * @param repository - The fhir repository
+     * @param converter - a model-entity to fhir-entity converter
+     */
+    @Autowired
+    public OrganizationStore(FhirRepository repository,
+            EntityConverter<Site, org.hl7.fhir.r4.model.Organization> converter) {
+        this.repository = repository;
+        this.converter = converter;
+    }
+
+    /**
+     *
+     * @param entity - The Site entity
+     * @return String
+     */
+    @Override
+    public String saveEntity(Site entity) {
+
+        // TODO: Add research study only when needed.
+        ResearchStudy researchStudyId = createResearchStudy();
```

Comment:
The order of things here is not right.
A research study is connect to an org, hence the org needs to be created first, and then create the research study and set its sponsor entity to point to the newly created org.